internal/experimental/registry/cache.go

```diff
@@ -39,451 +39,302 @@ import (
 	"helm.sh/helm/pkg/chartutil"
 )
 
-var (
-	tableHeaders = []string{"name", "version", "digest", "size", "created"}
+const (
+	CacheRootDir = "cache"
 )
 
 type (
-	filesystemCache struct {
-		out     io.Writer
-		rootDir string
-		store   *orascontent.Memorystore
+	// CacheOptions is used to construct a new cache
+	CacheOptions struct {
+		Debug   bool
+		Out     io.Writer
+		RootDir string
+	}
+
+	// Cache handles local/in-memory storage of Helm charts, compliant with OCI Layout
+	Cache struct {
+		debug       bool
+		out         io.Writer
+		rootDir     string
+		ociStore    *orascontent.OCIStore
+		memoryStore *orascontent.Memorystore
+	}
+
+	// CacheRefSummary contains as much info as available describing a chart reference in cache
+	// Note: fields here are sorted by the order in which they are set in FetchReference method
+	CacheRefSummary struct {
+		Name         string
+		Repo         string
+		Tag          string
+		Exists       bool
```

Comment:
Based on what I see in the code that uses this, should there just be a `ErrChartNotFound` or the like that can be used instead? Is there ever a case where it could not exist and things shouldn't error?

Reply:
Yes. Checkout client.go. This is used for example to print "Downloaded newer chart" vs "Chart is up to date", whether or not a chart existed in cache prior to to store operation